go/arrow/schema.go

```diff
@@ -197,6 +197,16 @@ func (sc *Schema) FieldIndices(n string) []int {
 	return sc.index[n]
 }
 
+// FieldIndex return index of the unique field with the given name.
```

Comment:
typo: `returns the index`

Also, is there a reason to have this in addition to the existing `FieldIndices` method? A consumer could just do the same length check and use index 0 themselves.